### PR TITLE
Editor: Surface debugging details in the ErrorBoundary

### LINKED ISSUE #34482: Improve editor error boundaries and error messages
## What problem does this address?
Whilst working through some issues on the forums lately, it's pretty clear the error boundaries in the editors  could be improved. Here's how it looks right now for reference:
Screenshot 2021-09-02 at 10 35 08 am

On the forums I've observed that users see someone else has posted a screenshot of this, and they'll post on the same thread saying they're experiencing the same bug. But this is actually just a generic error message and users most of the time are experiencing completely different issues.

Users also regularly fail to use the 'Copy Error' button, which is the most important part for developers.

'Copy Post Text' is fairly important for the user to keep their work, but that might not be so clear.

Finally, it's fairly common for multiple errors to be logged in the console, but the Copy Error button only copies the latest error message, which might not be entirely useful.

## What is your proposed solution?
While it's not clean or tidy, this UI should only display in exceptional circumstances, and I think the most recent error message should always be visible so that the user can clearly tell if they're experiencing the same issue as someone else. Not necessarily the full stack-trace, but at least the error message.

Further error details could be available in a log that can be expanded using something like a `<details>` element - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details. It should be possible for the error boundary to capture all the errors and push them onto an array for display here (stack traces included).

'Attempt Recovery' almost never works because it recreates the same error. Maybe it could try an 'undo' before re-initialising everything.

And one last thing, whenever this error occurs, there's usually a fairly routine response on the forums (deactivate all plugins, then reactivate them one by one etc.). It'd be great to link the user to a help article that contains details of these steps, and if that doesn't help how to report the issue.

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Closes #34482.

The editor's error boundary showed a generic sentence and copied the raw `error.stack`, which contains mostly React internals and says nothing about where in the editor the crash occurred. React passes a second argument to `componentDidCatch` with a `componentStack` describing the component tree at the throw site, and the boundary was discarding it.

This captures that component stack and puts it to use in three places.

* The `editor.ErrorBoundary.errorLogged` action now receives React's error info object as a second argument, so error tracking tools can record it.
* The "Copy error" button now produces a Markdown report containing the message, the stack trace, the component stack, and the user agent, ready to paste into a bug report.
* When `SCRIPT_DEBUG` is on, a collapsible panel shows the message and the component stack directly in the UI, so a developer can identify the failing component without opening the console.

## Testing Instructions
1. Ensure `SCRIPT_DEBUG` is enabled.
2. Fake an error in the editor. See testing patch below.

<details><summary>Testing patch</summary>
<p>

```diff
diff --git a/packages/edit-post/src/components/layout/index.js b/packages/edit-post/src/components/layout/index.js
index 650683f3a0f..1b6da9a7f94 100644
--- a/packages/edit-post/src/components/layout/index.js
+++ b/packages/edit-post/src/components/layout/index.js
@@ -346,6 +346,29 @@ function MetaBoxesMain() {
 	);
 }
 
+// TEMP: remove. Stands in for a plugin panel that reads the store and throws
+// while rendering. Nested so the component stack has something to show.
+function TempCrashInner( { title } ) {
+	if ( title.toLowerCase().includes( 'crash' ) ) {
+		throw new TypeError(
+			`Cannot read properties of undefined (reading 'length')`
+		);
+	}
+
+	return null;
+}
+
+// TEMP: remove. Type "crash" in the post title to trigger it.
+function TempCrash() {
+	const title = useSelect(
+		( _select ) =>
+			_select( editorStore ).getEditedPostAttribute( 'title' ) ?? '',
+		[]
+	);
+
+	return <TempCrashInner title={ title } />;
+}
+
 function Layout( {
 	postId: initialPostId,
 	postType: initialPostType,
@@ -556,6 +579,8 @@ function Layout( {
 			<Tooltip.Provider>
 				<ThemeProvider isRoot color={ { primary: adminPrimary } }>
 					<ErrorBoundary canCopyContent>
+						{ /* TEMP: remove. */ }
+						<TempCrash />
 						<WelcomeGuide postType={ currentPostType } />
 						<div { ...navigateRegionsProps }>
 							<Editor
```

</p>
</details> 

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 

| Post Editor | Site Editor |
|--------|--------|
|CleanShot 2026-08-20 at 12 49 36|CleanShot 2026-08-20 at 12 57 46|

## Use of AI Tools

Assisted by Claude.

### CODE REVIEW

File: packages/editor/src/components/error-boundary/index.js [hunk 1]
```diff
@@ -55,24 +140,43 @@ class ErrorBoundary extends Component {
...
+					</EmptyState.Title>
+					<EmptyState.Description>
+						{ __(
+							'Reload the page to continue. Copy the error to report the problem.'
+						) }
+					</EmptyState.Description>
+					<EmptyState.Actions>
+						{ canCopyContent && (
+							<CopyButton text={ getContent }>
+								{ __( 'Copy contents' ) }
```
Comment (ramonjd) [inline packages/editor/src/components/error-boundary/index.js:175]: Wondering if this could be confused with "contents" of the error. Maybe it could be explicit, e.g., "Copy post content" or "editor content".

Or maybe we could tweak the description to explain the option. 🤷🏻

File: packages/edit-post/src/components/layout/index.js [hunk 2]
```diff
@@ -346,6 +346,29 @@ function MetaBoxesMain() {
...
+			`Cannot read properties of undefined (reading 'length')`
+		);
+	}
+
+	return null;
+}
+
+// TEMP: remove. Type "crash" in the post title to trigger it.
+function TempCrash() {
+	const title = useSelect(
```
Comment (ramonjd) [inline packages/edit-post/src/components/layout/index.js]: I had to set `title` to `null` to get things to crash. Might be just me.

File: packages/editor/src/components/error-boundary/index.js [hunk 3]
```diff
@@ -19,6 +20,48 @@ function getContent() {
...
+	}
+
+	if ( componentStack ) {
+		sections.push(
+			`#### Component stack\n\n\`\`\`\n${ componentStack.trim() }\n\`\`\``
+		);
+	}
+
+	sections.push(
+		`#### Environment\n\n- User agent: ${ window.navigator.userAgent }`
```
Comment (ramonjd) [inline packages/editor/src/components/error-boundary/index.js]: ❤️

File: packages/editor/src/components/error-boundary/index.js [hunk 4]
```diff
@@ -55,24 +140,43 @@ class ErrorBoundary extends Component {
...
+				<EmptyState.Root>
+					<EmptyState.Icon icon={ caution } />
+					<EmptyState.Title>
+						{ __(
+							'The editor has encountered an unexpected error.'
+						) }
+					</EmptyState.Title>
+					<EmptyState.Description>
+						{ __(
+							'Reload the page to continue. Copy the error to report the problem.'
```
Comment (ramonjd) [inline packages/editor/src/components/error-boundary/index.js]: Reloading might not work if the editor is crashing? What about a "Reload to try again" and/or "Exit to dashboard" buttons

**Review by ramonjd — COMMENTED** [review]
Thanks for getting this started. It's working well for me functionally speaking.

I keep fullscreen on so it made me ask myself whether we need to give users an in-editor option to navigate away in that situation. I know there's the `wpadminbar` so maybe too much?

Idea (not actionable!!): just throwing it out there, it might be cool to see the error straight away, e.g., 

Image

Not very user friendly though. Maybe it could be collapsible. (Show error details button or something)

File: packages/edit-post/src/components/layout/index.js (hunk 2 above)
Comment (Mamaduka) [reply packages/edit-post/src/components/layout/index.js]: Noting that I need to remove this before we land :D

File: packages/editor/src/components/error-boundary/index.js (hunk 4 above)
Comment (Mamaduka) [reply packages/editor/src/components/error-boundary/index.js]: I think we had those buttons, but removed them in the last iteration. Recovery rarely worked when the error reached this boundary.

**Comment by Mamaduka** [comment]
Thanks for having a look, @ramonjd!

I like the design suggestions, especially the error report. I've improved "Component Stack" logic on a different branch; I mostly wanted to focus on design and UX here, but the idea is similar.

**Comment by jasmussen** [comment]
Thanks for the work, thanks for the ping. 

This has been in need for some love for a long time, honestly, but also some rethinking. But before I can answer fully, I need to better understand the details. One of the things about the error boundary is that it is literally a boundary: everything between A and B is broken, show something in its place. And in the old designs, this boundary could also be for a single block:

image

I know this is different, but that's context for why both cases used the black border: it was understood to be much more compartmentalised than I think in effect it turned out to be. Because coming at this today, I like to think of it in two ways:

1. A block is broken (example above).
2. The editor is broken.

Those are two entirely different things, and should be visually different as well, in the sense that if the whole editor is broken, there's no reason to pretend any blocks are still present. We're effectively designing a **crash screen** instead of an error boundary.

Because of that, my instinct (but I'm curious what Francisco thinks if he chimes in) would be to show a [Notice](https://wordpress.github.io/gutenberg/?path=/story/design-system-components-notice--error), followed by a [CollapsibleCard](https://wordpress.github.io/gutenberg/?path=/docs/design-system-components-collapsiblecard--docs), both of which have been designed to be able to sit next to each other. Something like:

```
┌───────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ The editor has crashed                                                                                │
│                                                                                                       │
│ An unknown  error occurred. Reload to try again, or copy the error to report the problem or search.   │
└───────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌───────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ Show error details                                                                                 ⌄  │
└───────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

You'd want that somewhere in the centre of the white crashed page, not too wide. 

What do you think?

File: packages/editor/src/components/error-boundary/index.js [hunk 5]
```diff
@@ -28,25 +71,67 @@ function CopyButton( { text, children, variant = 'secondary' } ) {
...
 		this.state = {
 			error: null,
+			componentStack: null,
 		};
 	}
 
-	componentDidCatch( error ) {
-		doAction( 'editor.ErrorBoundary.errorLogged', error );
+	componentDidCatch( error, errorInfo ) {
+		this.setState( { componentStack: errorInfo?.componentStack } );
```
Comment (jsnajdr) [inline packages/editor/src/components/error-boundary/index.js:142]: For setting the state we have the `getDerivedStateFromError` static method. `componentDidCatch` should only do the "side effect" as it did before, i.e., running the action.

Comment (Mamaduka) [reply packages/editor/src/components/error-boundary/index.js:142]: Had to add this to set/get detailed error info; the `getDerivedStateFromError` only receives `error`.

I think we can use `React.captureOwnerStack`, when we finally upgrade to React 19.

Comment (jsnajdr) [reply packages/editor/src/components/error-boundary/index.js:142]: You're right, even the React sources have an internal test/example that uses this same pattern: set the `error` state in `getDerivedStateFromError`, the `errorInfo` state manually from `componentDidCatch`.

**Comment by Mamaduka** [comment]
I'll try to refresh the design based on the feedback.

@jasmussen, just wanted to clarify something regarding Notice and its actions:

* "Reload to try again" - I assume this suggests a browser reload and not a specific action via a button. Right?
* Are we keeping the "Copy contents" action?
* Should I try to match the design for error details to what @ramonjd shared?

**Comment by jasmussen** [comment]
> "Reload to try again" - I assume this suggests a browser reload and not a specific action via a button. Right?

Yes, I'm trying carefully to make the error message more "friend to friend" and short, but perhaps that was too short. "Reload your browser to try again" might be better after all.

> Are we keeping the "Copy contents" action?

No strong opinion. If it sits inside the collapsiblecard, you're already engaging with "complexity", so it's fine. Is it actually useful? Relatedly a standard for code blocks appears to be slowly standardising on a small copy button in the top right corner of the code block itself. That's not an action item, we should probably keep this focused and simple, but something to ponder as a general question for error logs.

> Should I try to match the design for error details to what @ramonjd shared?

My apologies, perhaps I had not had enough coffee when I wrote my comment, somehow I glossed over Ramon's screenshot a bit too quickly. And yes there's a lot to like there, notably around the friend-to-friend phrasings. I wonder if there's some middleground, though, that has less text? 

The thing is, and this is purely anecdotal so I defer to you all: whenever I've seen this crash screen myself, I'm SOL. There's nothing I can really do but try the npm install/npm run build dance again. But I recognise that's very dev-oriented, so I actually have little instincts of how often this particular screen is seen by our users, or under which circumstances they will see it.

If the proposed steps are genuinely likely to be helpful to users, definitely include them. But let's not be exhaustive just for the sake of it. Is that helpful?

**Comment by jsnajdr** [comment]
When testing I see this, there is a lot of free space on the right:

Screenshot 2026-08-19 at 12 34 44

There  is an element with `editor-error-boundary` class, it defines `max-width: 780px`, and inside there is `EmptyState.Root`, whose `.root` style says `max-width: var(--wpds-dimension-surface-width-sm)` and that's 320px.

Is `EmptyState` a good component to use here? There is no empty state semantics.

In the error details, if we are showing the "Component Stack", can we also show the regular "Stack", i.e., where the error really happened, available in `error.stack`?

Does this need to be gated by `SCRIPT_DEBUG`? It's useful even on production sites, a fast way to see the error. The `componentStack` info should be available in production, too, it's not a dev-only feature.

As a followup, it would be nice if the `BlockCrashBoundary` also supported this, currently it says basically nothing about the error:

Screenshot 2026-08-19 at 12 43 19

**Comment by Mamaduka** [comment]
@jasmussen, oh, sorry, I should've clarified better.

ErrorBoundary (trunk) actions:

- "Copy contents" - copies editor contents (blocks) to avoid loss.
- "Copy error" - copies the error caught by the boundary. Useful for sharing errors for debugging.

ErrorBoundary (this branch) actions:

_It's also important to note that the "Show error details" toggle will be hidden in production. This is common practice._

* Do we want to retain existing actions as they are? 
* Do we want a second "Copy as markdown" in the toggle area? "Copy error" will copy markdown-formatted text. Easily pasteable in the issues or LLM chats. This action will be available in production.

@jsnajdr, right. I can't design to save my life, so I shared an ugly state to get better suggestions from @jasmussen, @fcoveram 😄 

> In the error details, if we are showing the "Component Stack", can we also show the regular "Stack", i.e., where the error really happened, available in error.stack?

Yes, we can share as many details as we want from available data.

> Does this need to be gated by SCRIPT_DEBUG? It's useful even on production sites, a fast way to see the error. The componentStack info should be available in production, too, it's not a dev-only feature.

I based this on practice used in other frameworks. WP also hides error info in production.

> As a followup, it would be nice if the BlockCrashBoundary also supported this, currently it says basically nothing about the error:

Once we've established a design pattern, I'll update the remaining boundaries to match it.

**Comment by Mamaduka** [comment]
Pushed some changes. Now the error uses the Notice component; error details show more info and match the design/colors shared by @ramonjd. I've also tried old grey colors (see screenshots).

I've not included any extra parsing for component stack, we can try that in follow-up.

**Comment by jasmussen** [comment]
I assume you updated the screenshots in the post, nice. Did you consider using the CollapsibleCard component for the error message, instead of the "Show error details" button?

**Comment by Mamaduka** [comment]
@jasmussen, here's how `CollapsibleCard` looks. All this nesting and these borders make the error boundary surface rather busy, IMO.

## Screenshot
CleanShot 2026-08-19 at 17 11 26

**Comment by jasmussen** [comment]
Yep, that's what I imagined, only without the black border. Do we need that at all on the _editor_ crash? When it's a _block_ crash the black border is there to be similar to that of a Placeholder.

**Comment by jasmussen** [comment]
Oh, I'd also start with the error collapsed by default.

**Comment by Mamaduka** [comment]
> Oh, I'd also start with the error collapsed by default.

I expanded it only for the screenshot.

> Yep, that's what I imagined, only without the black border. Do we need that at all on the editor crash? When it's a block crash the black border is there to be similar to that of a Placeholder.

I can drop the main container border. Are we keeping the shadow?

Here's how it looks without a border and in the default collapsed state.

CleanShot 2026-08-19 at 17 24 20

**Comment by jasmussen** [comment]
I'd keep only the notice and the collapsiblecard below. Any other containers and their borders and shadows, nor worth keeping, IMO. I think it's worth a shot! Thank you.

**Comment by Mamaduka** [comment]
Pushed changes and updated the screenshot in the description.

**Comment by jasmussen** [comment]
Nice. I assume that the collapsiblecard is opened for the purposes of the screenshot, per your previous note. How do you feel about it? 

One note: the dark background on the codeblock stands out quite a bit to the light surroundings. I'd expect that colour in dark mode, but not in what I suppose is effectively light mode for the editor at present. Can we limit the code block CSS to as much inheritance as possible? Do we need to apply overflow or can we just allow the CollapsibleCard to contain the full error however long it is? If we can do without the overflow, we could remove both the padding, and the background colour entirely, limiting the CSS for the code block to just the font and the wrapping. If we have to keep the overflow, I'd use the lightest grey you can find, with default text colour of the code inside.

**Comment by Mamaduka** [comment]
> Nice. I assume that the collapsiblecard is opened for the purposes of the screenshot, per your previous note. How do you feel about it?

Feels better, but also not very happy with the colors (like you noted). I guess there are too many things that catch the eye. Should we try a light grey background and default text color?

Error details can be quite large; we should keep the overflow.

**Comment by jasmussen** [comment]
Yep, I'd try keeping everything light, and remove as much custom CSS as you possibly can. 

I realise I'm giving live feedback here rather than just sharing a mockup, sorry—it's been a long day. If it helps, I can set aside some time tomorrow to get this right. Let me know!

**Comment by Mamaduka** [comment]
> I realise I'm giving live feedback here rather than just sharing a mockup, sorry—it's been a long day. If it helps, I can set aside some time tomorrow to get this right. Let me know!

Yes, that would be very helpful. Thanks, @jasmussen!

**Comment by jasmussen** [comment]
Here's my instinct. Collapsed:

crash-screen, collapsed

Expanded with no scrolling, just the full message:

crash-screen, expanded

If we have to have a scrollbar:

crash-screen, expanded scroll

The text is mono, 12px neutral-weak. 

Let me know if this is useful!

**Comment by Mamaduka** [comment]
Thanks, Joen!

* What `max-width` are you using for the container?
* I like the error details style, but the color contrast (light grey text on white) might not pass a11y guidelines.

> If we have to have a scrollbar

Yes, we have to; otherwise, the user might see only partial error details.

**Comment by jasmussen** [comment]
> What max-width are you using for the container?

540, but that was slightly arbitrary. I feel like we have some query presets somewhere, we should use one of those, whichever feels right.

> I like the error details style, but the color contrast (light grey text on white) might not pass a11y guidelines.

`neutral-weak` (which translates to 707070) passes 4.95:1 AA against white (and is designed to, it's for help-text). However it's possible I used the token incorrectly. If that's the case feel free to just inherit whatever black/dark/gray-900/neutral colour is inherited. The lightness of this colour is less about design-intent, and more about token-usage, so I defer to you on final choice.

**Comment by Mamaduka** [comment]
Pushed updates. Only minimal custom CSS (can't be avoided), using WP design tokens.

* Used `max-width: 560px` based on our dimensions system.
* Color contrast looks better in practice and, as Joen said, should pass an a11y check.
* I've included screenshots for both the Post and Site editors. The site editor has a dark background, and our error boundary adapts to it, thanks to new UI components and design tokens.

**Comment by jasmussen** [comment]
Looks good to me. I softly feel like both copy buttons should be secondary, not sure if the Notice allows this or not, so not a blocker. But IMO this is good to me. How does it feel to you?

**Comment by jsnajdr** [comment]
Some feedback:
- the "Stack" info should be before the "Component stack" info. "Stack" is the primary error stack, the component one is auxiliary and available only on some errors.
- visual design-wise, it seems odd that the entire screen is in an unified 13px font, except the "Error details" header that is 15px. It's a `wp-ui`/WPDS thing, a `CollapsibleCard.Heading` component. But it's odd that this is the only part of the UI that has a "heading" look, because it doesn't have a "heading" meaning.

> The text is mono, 12px neutral-weak.

This is something that is not visible to a naked eye: the notice and the headings are 13px (`font-size-md`) while the text is 12px (`font-size-sm`). To the user this distinction is meaningless, because it's practically invisible, it's more like an inconsistency, using two font sizes instead of one.

**Review by jasmussen — APPROVED** [review]

**Comment by jasmussen** [comment]
> This is something that is not visible to a naked eye: the notice and the headings are 13px (font-size-md) while the text is 12px (font-size-sm). To the user this distinction is meaningless, because it's practically invisible, it's more like an inconsistency, using two font sizes instead of one.

The motivation is that help-text is generally neutral-weak and 12px. It happens that optically the monospaced font is larger than the regular one, so it's better balanced.

However that's not critical for this juncture. If you all agree to use an inherited 13px font, I'd encourage just removing the font size override. Though if you do that, probably remove also the colour override, so that's inherited too, as it then moves away from the help-text connection.

**Comment by jsnajdr** [comment]
> It happens that optically the monospaced font is larger than the regular one

That's fair, my old eyes didn't recognize at first there is a monospace font 🙂 The Apple SF sans-serif and Menlo monospace fonts don't look that different at this size.

**Comment by Mamaduka** [comment]
> the "Stack" info should be before the "Component stack" info. "Stack" is the primary error stack, the component one is auxiliary and available only on some errors.

I can do the swap.

Don't have a strong opinion about font sizes. Though I like that error stack details are distinct from the body text.

**Comment by jasmussen** [comment]
I don't either have a strong opinion, but an argument for moving towards 13px black text is _less CSS_, and that's always good. Up to you all!

**Comment by Mamaduka** [comment]
> but an argument for moving towards 13px black text is less CSS, and that's always good. Up to you all!

Two lines of CSS don't change that much 😄

**Comment by jsnajdr** [comment]
> but an argument for moving towards 13px black text is less CSS

My typography education says that fewer font sizes are typically better, and that the old masters didn't have that many physical type sets anyway 🙂

**Review by jsnajdr — APPROVED** [review]
After the stack sections got swapped, my main concern is addressed and I think this is ready to ship 🚢

**Comment by Mamaduka** [comment]
* Swapped order in https://github.com/WordPress/gutenberg/pull/81642/commits/5d0ea1f26afc75e696ef879937a951a456c25341.
* Removed custom text styling in https://github.com/WordPress/gutenberg/pull/81642/commits/80a0c405936fcceb0026511f8322f5a0444a4880. Now it matches inherited styles as suggested.

## Screenshot

CleanShot 2026-08-20 at 14 11 31

**Comment by jasmussen** [comment]
Nice work, thanks for doing all this. 

While I'm here, can we use similar componentry for solving #62737? Or does that require a different approach?

**Comment by Mamaduka** [comment]
Technically, yes. We could use Notice with a warning variant; the component also has a speak message prop, which should resolve the issue mentioned.

One problem we might encounter here is that `MultipleUsageWarning` and the `Warning` block component are rendered inside the iframe canvas. There might be a slight chance that theme styles are affecting the components.

Here's a screenshot from a quick refactoring. I can create a PR to test tomorrow; I just need to wire up the speak for notices.

CleanShot 2026-08-20 at 18 31 01

**Comment by jasmussen** [comment]
> One problem we might encounter here is that MultipleUsageWarning and the Warning block component are rendered inside the iframe canvas. There might be a slight chance that theme styles are affecting the components.

Ah good catch. Also, when I test this myself, the sibling inserters sit in weird places as if the present error boundary is not actually rearranging content but is simply pushing it around. Whether we keep the Placeholder element for an error, that seems like the main thing to solve.

**Comment by jsnajdr** [comment]
> One problem we might encounter here is that `MultipleUsageWarning` and the `Warning` block component are rendered inside the iframe canvas. There might be a slight chance that theme styles are affecting the components.

It's very common for blocks to show placeholder UI directly inside the editor canvas: Image, Cover, Query Loop, Columns, ... These have potentially the same problem with theme styles, but we do it anyway. Either it's solved somehow (probably not) or it's a very minor problem easy to ignore.

Screenshot 2026-08-21 at 8 11 31

**Comment by andrewserong** [comment]
> These have potentially the same problem with theme styles, but we do it anyway. Either it's solved somehow (probably not) or it's a very minor problem easy to ignore.

From memory we used to run into conflicts from themes from time to time and have to fix gaps (e.g. back in https://github.com/WordPress/gutenberg/pull/58323), but we've got pretty good coverage of resets / specificity so it works pretty well nowadays. I reckon a similar approach for errors in the canvas would be okay (a bunch of resets)?

https://github.com/WordPress/gutenberg/blob/0cf7ede6504680579f4cdb20de25f58adf4eed5b/packages/components/src/placeholder/style.scss#L5-L42

**Comment by jasmussen** [comment]
An error in a Placeholder may be best, since it's in-canvas. It seems like the bigger issues to address on that other issue is the fact that it seems to mess with the flow, and the sibling inserters.
